### ISSUE TITLE
Not a valid VPC endpoint ID?

### ISSUE THREAD

**Issue by leason** [aws/chalice#2005]
Hi all, upgrading an older project after it sat idle for a while due to lambda 3.6 getting deprecated.  I updated to Python 3.9 and went ahead and updated to Chalice 1.27 while I was at it.

I finally got my dev and staging stages to deploy properly and show the right runtime in the AWS console, but then noticed my VPC info was getting cleared out on every deploy.  Turns out we now have native support for VPCs, which is great as my function has to connect to an RDS database which is running in a VPC (I think that's still best practice according to the docs).  

I'm able to manually update my function to point at the VPC, but when I run chalice deploy I get an error stating the VPCE I'm putting in is not valid.  I've triple checked I'm pasting in the right ID.  I ran in debug mode but don't see anything obvious - boto just spits out that error and Chalice bubbles it up.  

Any pointers on what I should be checking on?

**Comment by leason** [comment]
This morning I tried deleting the "staging" stage of the API and re-deploying to see if that helped.  The same error occurs but the VPC is properly configured in the AWS console for the function.  But since the deploy stops with that error, the API gateway isn't configured so the API is not functional.  

I'm really stumped with this.  The VPC endpoint is clearly valid as it's getting set up right, but something in the API response is not working for Chalice.  Maybe I need to manually update some other dependent library like boto?

**Comment by leason** [comment]
Ok, figured it out.  Turns out I didn't have (or need) a private VPCE set up.  If you simply specify the subnet IDs and security group IDs in the config the VPC connection will be made automatically.  My VPC wasn't configured in a way that would work as a private endpoint.  So the failure was actually happening on the AGI Gateway step.  I was able to replicate that in the console by trying to manually create an API gateway - My VPCE wasn't allowed to be used when configuring a private API.
